### PR TITLE
Update dependencies and remove outdated ones

### PR DESCRIPTION
**Pull request by mohe2015**
This time I started from scratch instead of trying to edit the old files. This makes the diff look a little bit ugly (especially karma.conf.js)
I hope this works better

Things to consider:
- [x] Does it work with older versions of nodejs? [ Except of v6.16.0 which should be fine as that's really old ]
- [x] TODO Add circle.yml back
* Apply eslint in other pullrequest?

### CODE REVIEW

**Comment by henrikingo** [comment]
Hi @mohe2015 

Original message still contained TODOs, but is this now in a shape where you'd like me to test this and have a look?

**Comment by mohe2015** [comment]
Yes. @henrikingo

**Comment by mohe2015** [comment]
ping @henrikingo. I'm sorry if you didn't have time, just want to make clear you didn't forget it.

File: karma.conf.js [hunk 1]
```diff
@@ -1,80 +1,97 @@
...
+      Chrome_without_security: {
+        base: 'ChromeHeadless',
+        flags: ['--no-sandbox']
       }
     },
 
-    // If browser does not capture, or produce output, in given timeout [ms], kill it
-    captureTimeout: 60*1000,
-    browserNoActivityTimeout: 60*1000,
-
```
Comment (henrikingo) [inline karma.conf.js]: Why are the timeouts no longer needed? I can see the tests passing, just want to learn and understand.

File: package.json [hunk 2]
```diff
@@ -22,26 +22,21 @@
     "url": "https://github.com/impress/impress.js/issues"
   },
   "scripts": {
-    "build": "node build.js",
-    "lint": "jshint src test/*.js && jscs src test/*.js",
-    "test": "karma start --single-run",
-    "test:dev": "karma start",
-    "test:sauce": "karma start karma.conf-sauce.js"
+    "build": "node build.js && terser --compress --mangle --comments '/^!/' --source-map --output js/impress.min.js js/impress.js",
```
[the remark this answers is not in the document]
Comment (mohe2015) [reply package.json]: I changed that, but I don't think it really looks cleaner. (See the commit)

File: qunit_test_runner.html [hunk 3]
```diff
@@ -6,22 +6,20 @@
 <html>
 <head>
   <meta charset="utf-8">
-  <title>QUnit tests for impress.js</title>
-  <link rel="stylesheet" href="node_modules/qunitjs/qunit/qunit.css">
+  <meta name="viewport" content="width=device-width">
+  <title>QUnit Example</title>
```
[the remark this answers is not in the document]
Comment (mohe2015) [reply qunit_test_runner.html]: Yeah, I just forgot to change it while copying the template from the official docs

File: src/impress.js [hunk 4]
```diff
@@ -15,6 +15,7 @@
  *  url:     http://impress.js.org
  *  source:  http://github.com/impress/impress.js/
  */
+/*! Licensed under MIT and GPL - http://github.com/impress/impress.js */
```
[the remark this answers is not in the document]
Comment (mohe2015) [reply src/impress.js]: Yeah, this line is still added to the minified version

File: karma.conf.js (hunk 1 above)
Comment (mohe2015) [reply karma.conf.js]: I just started with the official template. Therefore lots of things are reordered or removed in that file.

Comment (mohe2015) [reply karma.conf.js]: btw there are also sane defaults (http://karma-runner.github.io/3.0/config/configuration-file.html)

**Comment by mohe2015** [comment]
> Thanks a lot for doing this update!
> 
> Upgrading the linter causes lots of linting errors. I suggest you simply revert to the old linting for now, then send a separate PR that upgrades to eslint + makes all the fixes too.
> 
> Didn't yet read or test the circleci changes, but other than that this seems pretty good. Thanks again for helping to keep tooling up to date.

I think I addressed all requested changes. Thank you for this awesome project!

**Comment by mohe2015** [comment]
ping @henrikingo

File: README.md [hunk 5]
```diff
@@ -26,6 +26,19 @@ Note: For a minimal checkout, omit the `--recursive` option. This will leave out
 New features and fixes are continuously merged into the master branch, which is what the above command will check out. For the latest stable release, see the [Github Releases page](https://github.com/impress/impress.js/releases).
 
 
+## Development
```
Comment (henrikingo) [inline README.md]: Can you move this section down, into the "WANT TO CONTRIBUTE" header.

File: build.js [hunk 6]
```diff
@@ -48,4 +49,18 @@ html += '</body>\n</html>'
 
 var filename = path.resolve(__dirname, 'examples', 'index.html');
 fs.writeFileSync(filename, html);
-console.log(filename);
+
```
Comment (henrikingo) [inline build.js]: Thanks. I definitively prefer this.

Can you group this together with creation of js/impress.js: Move this above line 37.

File: package.json [hunk 7]
```diff
@@ -24,24 +24,22 @@
   "scripts": {
     "build": "node build.js",
     "lint": "jshint src test/*.js && jscs src test/*.js",
-    "test": "karma start --single-run",
-    "test:dev": "karma start",
-    "test:sauce": "karma start karma.conf-sauce.js"
+    "new-lint": "eslint src test",
```
Comment (henrikingo) [inline package.json]: Yeah, that works as a path forward.

File: .circleci/config.yml [hunk 8]
```diff
@@ -0,0 +1,29 @@
+version: 2
```
Comment (henrikingo) [inline .circleci/config.yml]: Looked into this and realized CircleCI requires this upgrade. Thanks again for doing this.

**Comment by mohe2015** [comment]
I think I moved everything @henrikingo

**Pull request merged**